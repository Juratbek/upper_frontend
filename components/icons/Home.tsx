import { IIconProps } from './types';

export const HomeIcon = ({ color = 'black', width = 20, height = 20 }: IIconProps) => (
  <svg width={width} height={height} viewBox='0 0 24 24' fill='none'>
    <path
      d='M9 17V21.9997H5C3.89543 21.9997 3 21.1043 3 19.9997V10.0332C3 9.73801 3.13046 9.45786 3.35643 9.26786L10.7129 3.08228C11.4569 2.45664 12.5431 2.45665 13.2871 3.08228L20.6436 9.26786C20.8695 9.45786 21 9.73801 21 10.0332V20C21 21.1046 20.1046 22 19 22H15V17C15 13 9 13 9 17Z'
      fill={color}
    />
  </svg>
);
