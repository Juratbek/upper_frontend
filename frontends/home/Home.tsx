import { ApiErrorBoundary } from 'components';
import { Spinner } from 'components/lib';
import {
  DefaultLabel,
  LABEL_ID_PARAM,
  LoadMoreButton,
  NoArticle,
  PublishedArticle,
  UserLabels,
} from 'components/molecules';
import { Advertisement } from 'components/organisms';
import { useDevice, useUrlParams } from 'hooks';
import { FC } from 'react';
import { usePublishedArticlesList } from 'store/clients/published-article';
import { addAmazonBucketUrl } from 'utils/published-article';

export const HomePage: FC = () => {
  const { getParam } = useUrlParams();
  const { isDesktop } = useDevice();
  const label = (getParam(LABEL_ID_PARAM) ?? DefaultLabel) as string;
  const {
    fetchNextPage,
    list: articles,
    hasNextPage,
    ...articlesRes
  } = usePublishedArticlesList(label);

  return (
    <>
      <UserLabels />
      {!isDesktop && <Advertisement />}
      <ApiErrorBoundary
        res={articlesRes}
        fallback={
          <div style={{ height: '15rem' }} className='content-center'>
            <Spinner />
          </div>
        }
      >
        {articles.length === 0 && !articlesRes.isLoading && <NoArticle label={label} />}
        {articles.map(addAmazonBucketUrl).map((article) => (
          <PublishedArticle article={article} key={article.id} />
        ))}
        {articles.length !== 0 && hasNextPage && (
          <LoadMoreButton
            onClick={() => fetchNextPage()}
            loading={articlesRes.isFetchingNextPage}
          />
        )}
      </ApiErrorBoundary>
    </>
  );
};
