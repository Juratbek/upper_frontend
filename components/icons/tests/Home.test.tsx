import { render, screen } from '@testing-library/react';

import { HomeIcon } from '../Home';

describe('Home icon', () => {
  it('renders the icon', () => {
    render(<HomeIcon />);
    const icon = screen.getByRole('icon');
    expect(icon).toMatchSnapshot();
  });
});
