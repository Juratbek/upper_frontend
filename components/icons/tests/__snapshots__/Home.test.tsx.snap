// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home icon > renders the icon 1`] = `
<svg
  fill="none"
  height="20"
  role="icon"
  viewBox="0 0 24 24"
  width="20"
>
  <path
    d="M9 17V21.9997H5C3.89543 21.9997 3 21.1043 3 19.9997V10.0332C3 9.73801 3.13046 9.45786 3.35643 9.26786L10.7129 3.08228C11.4569 2.45664 12.5431 2.45665 13.2871 3.08228L20.6436 9.26786C20.8695 9.45786 21 9.73801 21 10.0332V20C21 21.1046 20.1046 22 19 22H15V17C15 13 9 13 9 17Z"
    fill="black"
  />
</svg>
`;
